src/components/visualizations/chart/test/chartOptionsBuilder.spec.ts

```diff
@@ -298,34 +298,27 @@ describe('chartOptionsBuilder', () => {
                 type: 'column'
             };
 
-            it('should validate with "dataTooLarge: false" with category number less than default limit', () => {
-                const data = {
-                    series: Array(1),
-                    categories: Array(3).fill({
-                        name: 'Month',
-                        categories: Array(31)
-                    })
-                };
-                const validationResult = validateData(undefined, { ...chartOptions, data });
-                expect(validationResult).toEqual({
-                    dataTooLarge: false,
-                    hasNegativeValue: false
-                });
-            });
-
-            it('should validate with "dataTooLarge: true" with category number greater than default limit', () => {
+            it.each([
+                [
+                    'should validate with "dataTooLarge: false" with category number less than default limit',
+                    3,
+                    { dataTooLarge: false, hasNegativeValue: false }
+                ],
+                [
+                    'should validate with "dataTooLarge: true" with category number greater than default limit',
+                    31,
+                    { dataTooLarge: true, hasNegativeValue: false }
+                ]
+            ])('%s', (_testcaseDescription: string, categoriesNum: number, expectation: any) => {
```

Comment:
I think we should define a type for `expectation` variable and return type in `validateData` function as well

Reply:
it's done